mzLib/MassSpectrometry/PeakIndexing/IndexingEngine.cs

```diff
@@ -257,12 +258,17 @@ internal List<List<T>> GetBinsInRange(double mz, Tolerance ppmTolerance)
         }
 
         /// <summary>
-        /// Returns the peak that is closest to the target mz
+        /// <param name="charge"> an optional parameter used only for IIndexedMass and massIndexingEngine; must be null for mz peak indexing </param>
+        /// Returns the peak that is closest to the target mz from one bin
         /// </summary>
-        internal static T GetPeakFromBin(List<T> bin, double mz, int zeroBasedScanIndex, int peakIndexInBin, Tolerance ppmTolerance)
+        internal static T GetPeakFromBin(List<T> bin, double mz, int zeroBasedScanIndex, int peakIndexInBin, Tolerance ppmTolerance, int? charge = null)
         {
             T? bestPeak = default(T);
             if (peakIndexInBin < 0 || peakIndexInBin >= bin.Count) return bestPeak;
+            if (charge != null && bin.FirstOrDefault() is not IndexedMass)
```

Comment:
If the bin is empty and charge is not equal to null, this will throw an exception. I think you should check that bin.FirstOrDefault() != null